### PR TITLE
add leakyRelu Op

### PR DESCRIPTION
**Pull request by baominghelly**
# Description

We add a new Operator LeakyRelu in Cuda platform.

# Build && Test
1. build(Succeed): make install-python CUDA=ON 
```
Successfully built pyinfinitensor
Installing collected packages: pyinfinitensor
  Attempting uninstall: pyinfinitensor
    Found existing installation: pyinfinitensor 0.0.0
    Uninstalling pyinfinitensor-0.0.0:
      Successfully uninstalled pyinfinitensor-0.0.0
Successfully installed pyinfinitensor-0.0.0
```

2. unit test(Succeed): make test-onnx && make test-cpp. The failed test case isn't related to our change.
```

python3 pyinfinitensor/tests/test_onnx.py
import backend: <module 'backend' from '/home/libaoming/workplace/InfiniTensor/pyinfinitensor/src/pyinfinitensor/backend.cpython-310-x86_64-linux-gnu.so'>
................................model: resnet18-v2-7.onnx(44.64 MiB)
..........................
----------------------------------------------------------------------
Ran 58 tests in 3.188s

OK

99% tests passed, 1 tests failed out of 74

Total Test time (real) =  37.77 sec

The following tests FAILED:
         64 - test_cuda_rope (Failed)
Errors while running CTest
```

3. integ test(Succeed): python test_infinitensor.py --model=./simple_cnn.onnx --device=CUDA
```
import backend: <module 'backend' from '/home/libaoming/workplace/InfiniTensor/pyinfinitensor/src/pyinfinitensor/backend.cpython-310-x86_64-linux-gnu.so'>
arg setting:  Namespace(model='./simple_cnn.onnx', device='CUDA')

Model Name: simple
[1, 3, 32, 32]
The difference of results between ONNXRuntime and InfiniTensor looks good!
```

### CODE REVIEW

**Review by bitzyz — APPROVED** [review]

**Comment by bitzyz** [comment]
可以尝试加一下half数据类型的试试

File: pyinfinitensor/src/pyinfinitensor/onnx.py [hunk 1]
```diff
@@ -526,6 +526,17 @@ def __init__(
                     tensors[node.input[1]],
                     tensors.get(node.output[0]),
                 )
+            elif node.op_type == "LeakyRelu":
+
```
Comment (whjthu) [inline pyinfinitensor/src/pyinfinitensor/onnx.py:530]: 额外空行删掉

Comment (whjthu) [reply pyinfinitensor/src/pyinfinitensor/onnx.py:530]: 跟其他地方的格式保持一致

File: src/kernels/cuda/unary.cc [hunk 2]
```diff
@@ -155,6 +155,21 @@ class SoftmaxCudnn : public CudaKernelWithoutConfig {
     }
 };
 
+class LeakyReluCuda : public CudaKernelWithoutConfig {
+
```
Comment (whjthu) [inline src/kernels/cuda/unary.cc:159]: 额外空行

File: src/kernels/cuda/unary.cu [hunk 3]
```diff
@@ -143,6 +143,18 @@ __global__ void _cast_kernel(INPUT *input, OUTPUT *output, size_t n) {
     }
 }
 
+template <typename T>
+__global__ void _leaky_relu_kernel(T *input, T *output, size_t n,  float alphaValue) {
+
```
Comment (whjthu) [inline src/kernels/cuda/unary.cu:148]: 额外空行

File: src/kernels/cuda/unary.cu [hunk 4]
```diff
@@ -143,6 +143,18 @@ __global__ void _cast_kernel(INPUT *input, OUTPUT *output, size_t n) {
...
+    size_t index = threadIdx.x + blockIdx.x * blockDim.x;
+    size_t stride = blockDim.x * gridDim.x;
+    for (size_t i = index; i < n; i += stride) {
+
```
Comment (whjthu) [inline src/kernels/cuda/unary.cu:152]: 额外空行

File: src/kernels/cuda/unary.cu [hunk 5]
```diff
@@ -350,6 +362,16 @@ void cast_kernel(INPUT *input, OUTPUT *output, size_t num) {
         (input, output, num);
 }
 
+template <typename T>
+void leaky_relu_kernel(T *input, T *output, size_t num, float alphaValue) {
+
```
Comment (whjthu) [inline src/kernels/cuda/unary.cu:367]: 额外空行

File: src/kernels/cuda/unary.cu [hunk 6]
```diff
@@ -350,6 +362,16 @@ void cast_kernel(INPUT *input, OUTPUT *output, size_t num) {
...
+    int blocksize = block_work_size();
+    int gridsize = (num + blocksize - 1) / blocksize;
+    _leaky_relu_kernel<<<gridsize, blocksize, 0, CUDAStream::getCurrentStream()>>>
+    (input, output, num, alphaValue);
+
```
Comment (whjthu) [inline src/kernels/cuda/unary.cu:372]: 额外空行

**Review by whjthu — CHANGES_REQUESTED** [review]

File: pyinfinitensor/src/pyinfinitensor/onnx.py (hunk 1 above)
Comment (baominghelly) [reply pyinfinitensor/src/pyinfinitensor/onnx.py:530]: Done

File: src/kernels/cuda/unary.cc (hunk 2 above)
Comment (baominghelly) [reply src/kernels/cuda/unary.cc:159]: Done

File: src/kernels/cuda/unary.cu (hunk 3 above)
Comment (baominghelly) [reply src/kernels/cuda/unary.cu:148]: Done

File: src/kernels/cuda/unary.cu (hunk 4 above)
Comment (baominghelly) [reply src/kernels/cuda/unary.cu:152]: Done

File: src/kernels/cuda/unary.cu (hunk 5 above)
Comment (baominghelly) [reply src/kernels/cuda/unary.cu:367]: Done

File: src/kernels/cuda/unary.cu (hunk 6 above)
Comment (baominghelly) [reply src/kernels/cuda/unary.cu:372]: Done

**Review by whjthu — APPROVED** [review]

**Pull request merged**